spec/grover/processor_spec.rb

```diff
@@ -875,6 +875,16 @@
         it { expect(mean_colour_statistics(image)).to eq %w[0 0 255] }
       end
 
+      context 'when using remote browser option with HTML', remote_browser: true do
+        let(:options) { { browserWsEndpoint: 'ws://localhost:3000' } }
+        let(:url_or_html) { '<html><body style="background-color: blue"></body></html>' }
+
+        it { expect(convert.unpack('C*')).to start_with "\x89PNG\r\n\x1A\n".unpack('C*') }
+        it { expect(image.type).to eq 'PNG' }
+        it { expect(image.dimensions).to eq [800, 600] }
+        it { expect(mean_colour_statistics(image)).to eq %w[0 0 255] }
+      end
+
```

Comment:
I'm not convinced this is really proving that the WS endpoint stuff is actually working. If you were to revert the changes in the processor.cjs file these tests would still pass. Suggest moving remove this image based test, and instead use something like this up in the PDF test section:
```
        context 'when passing through launch params' do
          let(:options) { { 'browserWsEndpoint' => browser_ws_endpoint } }
          let(:browser_ws_endpoint) { 'ws://localhost:3000/' }
          let(:url_or_html) do
            <<-HTML
              <html>
                <body>
                  Speech recognition is <span id="test" />
                  <script type="text/javascript">
                    var speechSupported = "webkitSpeechRecognition" in window;
                    document.getElementById("test").innerHTML = speechSupported ? "supported" : "not supported"
                  </script>
                </body>
              </html>
            HTML
          end

          it { expect(pdf_text_content).to eq 'Speech recognition is supported' }

          context 'when WS endpoint param specifies disabling the speech API' do
            let(:browser_ws_endpoint) { 'ws://localhost:3000/?--disable-speech-api' }

            it { expect(pdf_text_content).to eq 'Speech recognition is not supported' }
          end
        end
```
That proves that it is the browser WS endpoint configuration that is changing the text content. Win!

Reply:
Hmm, I know they worked because I saw the logs from the docker container.
But I get your point, reverting the changes would still make this pass.

I'll try your suggestion when I next get to it, thanks.

Reply:
I added the suggested test as a new commit.
Indeed this one does fail when undoing the changes, while the others still pass.

Should I remove the other tests or should I keep them anyway?